v2/codetags/parser_test.go

```diff
@@ -29,100 +30,462 @@ func TestParse(t *testing.T) {
-			{Name: "n1", Value: "2", Type: ArgTypeInt},
-			{Name: "n2", Value: "-5", Type: ArgTypeInt},
-			{Name: "n3", Value: "0xFF00B3", Type: ArgTypeInt},
-			{Name: "n4", Value: "0o04167", Type: ArgTypeInt},
-			{Name: "n5", Value: "0b10101", Type: ArgTypeInt},
-		}},
-
-		// bools
-		{input: "bools(t: true, f:false)", expectKey: "bools", expectArgs: []Arg{
-			{Name: "t", Value: "true", Type: ArgTypeBool},
-			{Name: "f", Value: "false", Type: ArgTypeBool},
-		}},
-
-		// mixed type args
-		{input: "mixed(s: `value`, i: 2, b: true)", expectKey: "mixed", expectArgs: []Arg{
-			{Name: "s", Value: "value", Type: ArgTypeString},
-			{Name: "i", Value: "2", Type: ArgTypeInt},
-			{Name: "b", Value: "true", Type: ArgTypeBool},
-		}},
+		// Basic tag name tests
+		{
+			name:   "simple name",
+			input:  "name",
+			expect: mkt("name"),
+		},
+		{
+			name:   "name with comment",
+			input:  "name // comment",
+			expect: mkt("name"),
+		},
+		{
+			name:   "name with dash",
+			input:  "name-dash",
+			expect: mkt("name-dash"),
+		},
+		{
+			name:   "name with dot",
+			input:  "name.dot",
+			expect: mkt("name.dot"),
+		},
+		{
+			name:   "name with colon",
+			input:  "name:colon",
+			expect: mkt("name:colon"),
+		},
+
+		// String arguments tests
+		{
+			name:   "empty parentheses",
+			input:  "name()",
+			expect: mkt("name"),
+		},
+		{
+			name:   "single argument",
+			input:  "name(arg)",
+			expect: mkta("name", mksa("arg")),
+		},
+		{
+			name:   "argument with comment",
+			input:  "name(arg) // comment",
+			expect: mkta("name", mksa("arg")),
+		},
+		{
+			name:   "uppercase argument",
+			input:  "name(ARG)",
+			expect: mkta("name", mksa("ARG")),
+		},
+		{
+			name:   "mixed case argument",
+			input:  "name(ArG)",
+			expect: mkta("name", mksa("ArG")),
+		},
+		{
+			name:   "argument with dash",
+			input:  "name(has-dash)",
+			expect: mkta("name", mksa("has-dash")),
+		},
+		{
+			name:   "argument with dot",
+			input:  "name(has.dot)",
+			expect: mkta("name", mksa("has.dot")),
+		},
+		{
+			name:   "lowercase argument",
+			input:  "name(lower)",
+			expect: mkta("name", mksa("lower")),
+		},
+		{
+			name:   "uppercase argument",
+			input:  "name(CAPITAL)",
+			expect: mkta("name", mksa("CAPITAL")),
+		},
+		{
+			name:   "mixed case argument 1",
+			input:  "name(MiXeD)",
+			expect: mkta("name", mksa("MiXeD")),
+		},
+		{
+			name:   "mixed case argument 2",
+			input:  "name(mIxEd)",
+			expect: mkta("name", mksa("mIxEd")),
+		},
+		{
+			name:   "underscore in argument",
+			input:  "name(_under)",
+			expect: mkta("name", mksa("_under")),
+		},
+		{
+			name:   "quoted argument",
+			input:  `name("hasQuotes")`,
+			expect: mkta("name", mksa("hasQuotes")),
+		},
+		{
+			name:   "raw quoted argument",
+			input:  "name(`hasRawQuotes`)",
+			expect: mkta("name", mksa("hasRawQuotes")),
+		},
+		{
+			name:   "raw string with spaces",
```

Comment:
how about a test for `"true"` and `"false"` (maybe do one with backticks) to distinguish `string("true")` from `bool(true)` ?